.github/workflows/build-and-test.yaml

```diff
@@ -0,0 +1,134 @@
+      bundle:
+        type: string
+        default: ""
+      platform:
+        type: string
+        default: 'ubuntu-latest'
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  setup:
+    outputs:
+      bundle-path: ${{ steps.bundle-default.outputs.path }}
+      capability-path: ${{ steps.capability-default.outputs.path }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set Bundle Path
+        id: bundle-default
+        run: |
+          PATH="."
+          [ ${{ inputs.bundle }} != 'uds-core' ] && PATH="${{ inputs.bundle }}"
+          echo "path=$PATH" >> $GITHUB_OUTPUT   
+        if: ${{ inputs.bundle }} != ''
+      - name: Set Capability Path
+        id: capability-default
+        run: |
+          echo "path=capabilities/${{ inputs.capability }}" >> $GITHUB_OUTPUT   
+        if: ${{ inputs.capability }} != ''
+
+  build-and-test:
+    needs: setup
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      # - name: Login to Registry1
+      #   uses: docker/login-action@v3
+      #   with:
+      #     registry: registry1.dso.mil
+      #     username: ${{ secrets.REGISTRY1_USERNAME }}
+      #     password: ${{ secrets.REGISTRY1_PASSWORD }}
+      - name: Login to GHCR
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: dummy
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install Zarf
+        uses: defenseunicorns/setup-zarf@main
+        with:
+          # renovate: datasource=github-tags depName=defenseunicorns/zarf versioning=semver
+          version: v0.30.1
+          download-init-package: true
+      - name: Login to GHCR
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: dummy
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Use Node.js latest
+        uses: actions/setup-node@5e21ff4d9bc1a8cf6de233a3057d20ec6b3fb69d # v3.8.1
+        with:
+          node-version: latest
+
+
+      - name: Setup docker (missing on MacOS)
+        if: runner.os == 'macos'
+        run: |
+              brew install docker
+              colima start --cpu 2 --memory 12
+              sudo ln -sf $HOME/.colima/default/docker.sock /var/run/docker.sock          
+
+      - name: Install k3d
+        run: |
+          curl -s https://raw.githubusercontent.com/k3d-io/k3d/main/install.sh | bash
+
+      - name: Build Capability
+        shell: bash
+        run: zarf package create --confirm --no-progress
+        working-directory: "${{ needs.setup.outputs.capability-path }}/.github/zarf-runner/bob-the-builder"
+        if: ${{ inputs.capability }} != ''
+
+      - name: Deploy Capability
+        shell: bash
+        run: zarf package create --confirm --no-progress
+        working-directory: "${{ needs.setup.outputs.capability-path }}/.github/zarf-runner/deploy"
```

Comment:
confused by the working directory call here, seems pretty messy. wouldnt it make more sense (if this is necessary to pass into `zarf package create dir`?  Though not a huge fan of this format and want to talk through it more.

Reply:
If its confusing whats happening and why maybe we should have a conversation synchronously

Reply:
Which of these is more readable:
```

      - name: Build Capability
        shell: bash
        run: zarf package create --confirm --no-progress
        working-directory: "${{ needs.setup.outputs.capability-path }}/.github/zarf-runner/bob-the-builder"
        if: ${{ inputs.capability }} != ''

        - name: Build Capability
        shell: bash
        run: |
          zarf package create \
          "${{ needs.setup.outputs.capability-path }}/.github/zarf-runner/bob-the-builder" \
          --confirm --no-progress
        if: ${{ inputs.capability }} != '' 
```
        
  I think the first one. 
        
        
   but again, the path after the github var is an artifact of how we need to do things now while we are in a holding pattern about task-runner things

Reply:
We can address this once we have mvp of the task runner Jordan is back to working on.